src/test/java/com/google/firebase/FirebaseAppTest.java

```diff
@@ -419,6 +424,99 @@ public void testTokenRefresherStateMachine() {
     assertEquals(0, refresher.cancelCalls);
   }
 
+  @Test(expected = IllegalStateException.class)
+  public void testDefaultAppEmptyFile() {
+    setDefaultConfigEnv("firebase_config_empty.json");
+    FirebaseApp.initializeApp();
+  }  
+  
+  @Test
+  public void testDefaultAppEmptyJSONString() {
+    setDefaultConfigEnv("");
+    FirebaseApp firebaseApp = FirebaseApp.initializeApp();
+    assertEquals(null, firebaseApp.getOptions().getProjectId());
+    assertEquals(null, firebaseApp.getOptions().getStorageBucket());
+    assertEquals(null, firebaseApp.getOptions().getDatabaseUrl());
+    assertEquals(new HashMap<String, Object>(), 
+                 firebaseApp.getOptions().getDatabaseAuthVariableOverride());
+  }
+
+  @Test
+  public void testDefaultAppEmptyJSONObject() {
+    setDefaultConfigEnv("{}");
+    FirebaseApp firebaseApp = FirebaseApp.initializeApp();
+    assertEquals(null, firebaseApp.getOptions().getProjectId());
+    assertEquals(null, firebaseApp.getOptions().getStorageBucket());
+    assertEquals(null, firebaseApp.getOptions().getDatabaseUrl());
+    assertEquals(new HashMap<String, Object>(), 
+                 firebaseApp.getOptions().getDatabaseAuthVariableOverride());
+  }
+
+  @Test(expected = JsonSyntaxException.class)
+  public void testDefaultAppBadFile() {
+    setDefaultConfigEnv("firebase_config_bad.json");
+    FirebaseApp.initializeApp();
+  }
+
+  @Test(expected = JsonSyntaxException.class)
+  public void testDefaultAppBadJSON() {
+    setDefaultConfigEnv("{,,");
+    FirebaseApp.initializeApp();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testDefaultNoSuchFile() {
+    setDefaultConfigEnv("no_such.json");
+    FirebaseApp.initializeApp();
+  }
+
+  @Test
+  public void testDefaultGood() {
+    setDefaultConfigEnv("firebase_config.json");
+    FirebaseApp firebaseApp = FirebaseApp.initializeApp();
+    assertEquals("hipster-chat-mock", firebaseApp.getOptions().getProjectId());
+    assertEquals("hipster-chat.appspot.mock", firebaseApp.getOptions().getStorageBucket());
+    assertEquals("https://hipster-chat.firebaseio.mock", firebaseApp.getOptions().getDatabaseUrl());
+    assertEquals("authVal", firebaseApp.getOptions().getDatabaseAuthVariableOverride()
+                              .get("this#is#an#auth#string"));
+  }
+
+  @Test
+  public void testDefaultGoodIgnored() {
+    setDefaultConfigEnv("firebase_config.json");
+    FirebaseApp firebaseApp = FirebaseApp.initializeApp(OPTIONS);
+    assertEquals(null, firebaseApp.getOptions().getProjectId());
+    assertEquals(null, firebaseApp.getOptions().getStorageBucket());
+    assertEquals(null, firebaseApp.getOptions().getDatabaseUrl());
+    assertEquals(new HashMap<String, Object>(), 
+                 firebaseApp.getOptions().getDatabaseAuthVariableOverride());
+  }
+
+  @Test
+  public void testDefaultGoodJson() {
```

Comment:
testValidJson() -- We use valid/invalid instead of good/bad (revise throughout this file)